### PR TITLE
Fix less than operator for binary datatype

### PR DESCRIPTION
**Pull request by sumitj824**
### Description
Earlier, the less than operator for binary data type was pointing to incorrect function, leading to incorrect results and dump/restore failures for partitioned tables. This commit fixes the less than operator for binary datatype to point to correct function.

Signed-off-by: Sumit Jaiswal <sumiji@amazon.com>
### Issues Resolved

Task: BABEL-4957

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** NA

* **Arbitrary inputs -** Yes

* **Negative test cases -** NA

* **Minor version upgrade tests -** Yes

* **Major version upgrade tests -** Yes

* **Performance tests -** NA

* **Tooling impact -** NA

* **Client tests -** NA

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_common/sql/binary.sql [hunk 1]
```diff
@@ -249,7 +249,7 @@ CREATE OPERATOR sys.>= (
 
 CREATE FUNCTION sys.binary_lt(leftarg sys.bbf_binary, rightarg sys.bbf_binary)
 RETURNS boolean
-AS 'babelfishpg_common', 'varbinary_eq'
```
Comment (KushaalShroff) [inline contrib/babelfishpg_common/sql/binary.sql:252]: Wow, how did we not catch this?

**Review by KushaalShroff — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_common/sql/binary.sql (hunk 1 above)
Comment (sumitj824) [reply contrib/babelfishpg_common/sql/binary.sql:252]: We don't have operator-related test coverage for almost all data types.

**Review by shah-nirmit — APPROVED** [review]
LGTM

**Review by forestkeeper — DISMISSED** [review]
LGTM

**Review by KushaalShroff — APPROVED** [review]

**Pull request merged**